webapp/components/ContributionForm/ContributionForm.spec.js

```diff
@@ -356,19 +389,40 @@ describe('ContributionForm.vue', () => {
           postTitleInput = wrapper.find('.ds-input')
           postTitleInput.setValue(postTitle)
           wrapper.vm.updateEditorContent(postContent)
-          wrapper.find(CategoriesSelect).vm.$emit('updateCategories', categoryIds)
-          wrapper.find('.submit-button-for-test').trigger('click')
+          wrapper
+            .findAll('li')
+            .at(0)
+            .trigger('click') // language
+          await wrapper.find('form').trigger('submit')
           expect(mocks.$apollo.mutate).toHaveBeenCalledWith(expect.objectContaining(expectedParams))
         })
 
         it('supports updating categories', async () => {
-          const categoryIds = ['cat3', 'cat51', 'cat37']
+          expectedParams.variables.categoryIds = ['cat12', 'cat3', 'cat15']
           postTitleInput = wrapper.find('.ds-input')
           postTitleInput.setValue(postTitle)
           wrapper.vm.updateEditorContent(postContent)
-          expectedParams.variables.categoryIds = categoryIds
-          wrapper.find(CategoriesSelect).vm.$emit('updateCategories', categoryIds)
-          await wrapper.find('.submit-button-for-test').trigger('click')
+          wrapper
+            .findAll('li')
+            .at(0)
+            .trigger('click') // language
+          wrapper.find(CategoriesSelect).setData({ categories })
+          await wrapper
+            .find(CategoriesSelect)
+            .findAll('button')
+            .at(0)
+            .trigger('click')
+          await wrapper
+            .find(CategoriesSelect)
+            .findAll('button')
+            .at(3)
+            .trigger('click')
+          await wrapper
+            .find(CategoriesSelect)
+            .findAll('button')
+            .at(4)
+            .trigger('click')
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-11-12T20:36:57Z" title="Tuesday, November 12th 2019, 9:36:57 pm +01:00">Nov 12, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -356,19 +389,40 @@ describe('ContributionForm.vue', () => {
           postTitleInput = wrapper.find('.ds-input')
           postTitleInput.setValue(postTitle)
           wrapper.vm.updateEditorContent(postContent)
-          wrapper.find(CategoriesSelect).vm.$emit('updateCategories', categoryIds)
-          wrapper.find('.submit-button-for-test').trigger('click')
+          wrapper
+            .findAll('li')
+            .at(0)
+            .trigger('click') // language
+          await wrapper.find('form').trigger('submit')
           expect(mocks.$apollo.mutate).toHaveBeenCalledWith(expect.objectContaining(expectedParams))
         })
 
         it('supports updating categories', async () => {
-          const categoryIds = ['cat3', 'cat51', 'cat37']
+          expectedParams.variables.categoryIds = ['cat12', 'cat3', 'cat15']
           postTitleInput = wrapper.find('.ds-input')
           postTitleInput.setValue(postTitle)
           wrapper.vm.updateEditorContent(postContent)
-          expectedParams.variables.categoryIds = categoryIds
-          wrapper.find(CategoriesSelect).vm.$emit('updateCategories', categoryIds)
-          await wrapper.find('.submit-button-for-test').trigger('click')
+          wrapper
+            .findAll('li')
+            .at(0)
+            .trigger('click') // language
+          wrapper.find(CategoriesSelect).setData({ categories })
+          await wrapper
+            .find(CategoriesSelect)
+            .findAll('button')
+            .at(0)
+            .trigger('click')
+          await wrapper
+            .find(CategoriesSelect)
+            .findAll('button')
+            .at(3)
+            .trigger('click')
+          await wrapper
+            .find(CategoriesSelect)
+            .findAll('button')
+            .at(4)
+            .trigger('click')
```


maybe we don't need that many? don't know... I guess it's ok...